src/java.base/share/classes/jdk/internal/classfile/impl/StackMapGenerator.java

```diff
@@ -1058,23 +1055,26 @@ void setLocalsFromArg(String name, MethodTypeDesc methodDesc, boolean isStatic,
             }
             for (int i = 0; i < methodDesc.parameterCount(); i++) {
                 var desc = methodDesc.parameterType(i);
-                if (!desc.isPrimitive()) {
-                    setLocalRawInternal(localsSize++, Type.referenceType(desc));
-                } else switch (desc.descriptorString().charAt(0)) {
-                    case 'J' -> {
-                        setLocalRawInternal(localsSize++, Type.LONG_TYPE);
-                        setLocalRawInternal(localsSize++, Type.LONG2_TYPE);
+                if (desc == CD_void) throw new AssertionError("Should not reach here");
```

Comment:
You can use
```suggestion
                assert desc != CD_void;
```
So this fails in unit tests. But this takes code size.